src/client/QXmppMovedManager.cpp

```diff
@@ -0,0 +1,260 @@
+                                                                         // provided in the URI (e.g. user@example.com).
+                                                                         if (err.isStanzaError()) {
+                                                                             const auto e = std::any_cast<QXmppStanza::Error>(err.error);
+                                                                             const auto newJid = [&e]() -> QString {
+                                                                                 if (e.condition() != QXmppStanza::Error::Condition::Gone) {
+                                                                                     return {};
+                                                                                 }
+
+                                                                                 // TODO: Move Kaidan QXmppUri to QXmpp library
+                                                                                 QUrl url(e.redirectionUri());
+
+                                                                                 if (!url.isValid() || url.scheme() != u"xmpp") {
+                                                                                     return {};
+                                                                                 }
+
+                                                                                 return url.path();
+                                                                             }();
+
+                                                                             if (!newJid.isEmpty()) {
+                                                                                 return movedJidsMatch(newBareJid, newJid);
+                                                                             }
+                                                                         }
+
+                                                                         return err;
+                                                                     } },
+                                                                 std::move(result));
+                                           });
+}
+
+///
+/// Notify a contact we have moved
+///
+/// \param contactBareJid JID of the contact to send the subscription request
+/// \param oldBareJid JID of the old account we moved from
+/// \param reason The reason of the move
+///
+/// \return the result of the action
+///
+QXmppTask<QXmpp::SendResult> QXmppMovedManager::notifyContact(const QString &contactBareJid, const QString &oldBareJid, const QString &reason)
+{
+    QXmppPresence packet;
+    packet.setTo(QXmppUtils::jidToBareJid(contactBareJid));
+    packet.setType(QXmppPresence::Subscribe);
+    packet.setStatusText(reason);
+    packet.setOldJid(oldBareJid);
+    return client()->sendSensitive(std::move(packet));
```

Comment:
Whether encryption is used should be up to the user. Maybe that decision can be passed as an argument. @lnjX What do you think?